packages/plugins/sns/src/aws-sns/index.ts

```diff
@@ -0,0 +1,91 @@
+import { SNS, Credentials } from 'aws-sdk';
+
+class AwsSNS {
+  awsSNS: SNS
+
+  init() {
+    const SNSConfigOptions:SNS.ClientConfiguration = {
+      credentials: new Credentials(process.env['RLN_SNS_AWS_ACCESS_KEY_ID'], process.env['RLN_SNS_AWS_SECRET_ACCESS_KEY']),
+      region: process.env['RLN_SNS_AWS_REGION'],
+    };
+    this.awsSNS = new SNS(SNSConfigOptions);
+  }
+
+  /*
+  async sendEmail(
+    sender: string,
+    receivers: string[],
+    subject: string,
+    content: string,
+    awsSendEmailParams?: SES.Types.SendEmailRequest,
+  ): Promise<SES.Types.SendEmailResponse> {
+    const params: SES.Types.SendEmailRequest = {
+      Source: sender,
+      Destination: {
+        ToAddresses: receivers,
+      },
+      Message: {
+        Subject: {
+          Charset: 'UTF-8',
+          Data: subject,
+        },
+        Body: {
+          Html: {
+            Charset: 'UTF-8',
+            Data: content,
+          },
+        },
+      },
+      ...awsSendEmailParams,
+    };
+    return await this.awsSES.sendEmail(params).promise();
+  }
+   */
+
+  async createTopic(topicName: string, displayName: string = topicName) {
+    const params: SNS.Types.CreateTopicInput = {
+      Name: topicName,
+      Attributes:{
+        DisplayName: displayName,
+      },
+    };
+    return await this.awsSNS.createTopic(params).promise();
+  }
+
+  async publish(title: string, ARN: string, message: string, eventId: string){
+    const params: SNS.Types.PublishInput = {
+      MessageStructure: 'json',
+      Message: JSON.stringify({
+        default: 'Nuova Notifica',
+        APNS:{
+          aps: {
+            alert:{
+              title: title,
+              body: message,
+            },
+            sound: 'bingbong.aiff',
+          },
+          eventId: eventId,
+        },
+        GCM:{
+          data: {
+            title: title,
+            body: message,
+            vibrate: true,
+            eventId: eventId,
+          },
+        },
+      }),
+      TopicArn: ARN,
+    };
+    return await this.awsSNS.publish(params).promise();
+  }
```

Comment:
Remember this is a generic plugin, not just for Adhesion :-) so we can not make it too much strict.
It was better as it was before sorry :-) let's decide the project that installs the plugin which payload should be sent!
But this code will be useful project side

Reply:
oh yes, sorry, but that's the correct way to handle custom messages? Sice we don't want to messup AWS with tests we wanted to ask you if that's the correct way to handle SNS publishings.

Reply:
I have no idea, I sent them with the AWS dashboard :D we will know only when we'll try ;-) don't worry!